numpy/lib/function_base.py

```diff
@@ -115,13 +124,28 @@ def scott(x):
         Scott Estimator
         The binwidth is proportional to the standard deviation of the data and
         inversely proportional to the cube root of data size (asymptotically optimal)
-
         """
         h = 3.5 * x.std() * x.size ** (-1.0 / 3)
         if h > 0:
             return np.ceil(x.ptp() / h)
         return 1
 
+    def doane(x):
+        """
+        Doane's Estimator
+        Improved version of Sturges' formula which works better for non-normal
+        data.
+        See http://stats.stackexchange.com/questions/55134/doanes-formula-for-histogram-binning
+        """
+        if x.size > 2:
+            stdskew = np.sqrt(6.0 * (x.size - 2) / ((x.size + 1.0) * (x.size + 3)))
+            sigma = np.std(x)
+            if sigma > 0:
+                mu = np.mean(x)
+                skew = (np.mean(x**3) - 3 * mu * sigma**2 - mu**3) / sigma**3
```

Comment:
Normally yes, but that is not what the formula really does. Starting with the basic definition on the Histogram page at https://en.wikipedia.org/wiki/Histogram#Number_of_bins_and_width, which defines everything but <tt>|g<sub>1</sub>|</tt>. I took that from https://en.wikipedia.org/wiki/Skewness#Pearson.27s_moment_coefficient_of_skewness, where the formula <tt>kappa<sub>3</sub>/kappa<sub>2</sub><sup>3/2</sup></tt> eventually expands to the one I have used. Since a bunch of the trems in `(x-mu)^3` cancel out along the way, it is better not to subtract the mean off up front.